activerecord/test/cases/nested_attributes_test.rb

```diff
@@ -658,6 +658,16 @@ def test_should_raise_RecordNotFound_if_an_id_is_given_but_doesnt_return_a_recor
     assert_equal "Couldn't find #{@child_1.class.name} with ID=1234567890 for Pirate with ID=#{@pirate.id}", exception.message
   end
 
+  def test_should_raise_RecordNotFound_if_an_id_belonging_to_a_different_record_is_given
+    other_pirate = Pirate.create! catchphrase: 'Ahoy!'
+    other_child = other_pirate.send(@association_name).create! name: 'Buccaneers Servant'
+
+    exception = assert_raise ActiveRecord::RecordNotFound do
+      @pirate.attributes = { association_getter => [{ :id => other_child.id }] }
```

Comment:
no thats fine. We just change the syntax within the modified lines to keep the diffs focused.